src/middlewared/middlewared/plugins/device_/udev_events_linux.py

```diff
@@ -19,8 +19,16 @@ def udev_events(middleware):
     monitor = pyudev.Monitor.from_netlink(context)
     monitor.filter_by(subsystem='block')
     monitor.filter_by(subsystem='net')
-    for device in iter(monitor.poll, None):
-        middleware.call_hook_sync(f'udev.{device.subsystem}', data={**dict(device), 'SYS_NAME': device.sys_name})
+    while True:
+        # We always want to keep polling udev, let's log what error we are
+        # seeing and fix them accordingly as we see them
+        try:
+            for device in iter(monitor.poll, None):
+                middleware.call_hook_sync(
+                    f'udev.{device.subsystem}', data={**dict(device), 'SYS_NAME': device.sys_name}
+                )
+        except Exception:
+            middleware.logger.error('Polling udev failed', exc_info=True)
```

Comment:
Could we run into an infinite busy loop here?

e.g. monitor.poll for some reason always raise an exception like EOFError or StopIteration

Perhaps we should add a sleep here